syntax = "proto3";

package hipstershop;

// 评论服务
service CommentService {
    // 添加评论
    rpc AddComment(AddCommentRequest) returns (AddCommentResponse) {}
    // 获取评论
    rpc GetComments(GetCommentsRequest) returns (GetCommentsResponse) {}
}

// 添加评论请求
message AddCommentRequest {
    string product_id = 1;
    string user_id = 2;
    string comment_text = 3;
    int32 rating = 4;
}

// 添加评论响应
message AddCommentResponse {
    bool success = 1;
    string comment_id = 2;
}

// 获取评论请求
message GetCommentsRequest {
    string product_id = 1;
}

// 评论对象
message Comment {
    string comment_id = 1;
    string product_id = 2;
    string user_id = 3;
    string comment_text = 4;
    int32 rating = 5;
    string created_at = 6;
}

// 获取评论响应
message GetCommentsResponse {
    repeated Comment comments = 1;
} 